compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -311,6 +311,7 @@ impl<'context> Elaborator<'context> {
         let old_function = std::mem::replace(&mut self.current_function, Some(id));
 
         // Without this, impl methods can accidentally be placed in contracts. See #3254
+        let was_in_contract = self.in_contract;
```

Comment:
`self.in_contract` is reset to its previous value of `was_in_contract` only at the end of this function so that the change done by the if statement doesn't affect the next functions that are elaborated

Reply:
Ah yes, ok got it.